install/firmware.sh.in

```diff
@@ -0,0 +1,104 @@
+#!/bin/sh
+# vim: et sts=4 sw=4
+
+set -eu
+
+firmware="#FIRMWARE#"
+terms_url="https://www.microsoft.com/en-us/legal/terms-of-use"
+
+accept_terms=0
+
+usage() {
+    echo
+    echo "Usage: $(basename "$0") [OPTIONS]"
```

Comment:
```suggestion
    echo "Usage: xow-get-firmware.sh [OPTIONS]"
```

Is there any reason why we can't set this to `xow-get-firmware.sh` instead of using `basename`?

Reply:
Hard-coding the name is considered anti-pattern across the board - be that shell/python scripts or compiled languages.

Reply:
Okay, makes sense. I was just wondering because I've looked at the GNU coreutils and each individual program has its name defined directly in the source code, e.g.:

https://github.com/coreutils/coreutils/blob/2ba6fb653c7d029e9ecef8e6d7d762674d2573ff/src/cat.c#L45

Reply:
Not quite

coreutils uses `program_name`
https://github.com/coreutils/coreutils/blob/2ba6fb653c7d029e9ecef8e6d7d762674d2573ff/src/cat.c#L94

and sets it by calling `set_program_name(argv[0]);`
https://github.com/coreutils/coreutils/blob/2ba6fb653c7d029e9ecef8e6d7d762674d2573ff/src/cat.c#L592

those two come from gnulib's progname
which sets program_name to a modified variant of what has been passed to the function
https://github.com/coreutils/gnulib/blob/3c99da63be95a3bc6cc881e1ca6c849a921569e1/lib/progname.c#L84

```
$ ln -s $(which cat) bla
$ ./bla --help
Usage: ./bla [OPTION]... [FILE]...
Concatenate FILE(s) to standard output.
```

Reply:
Oh, I see. Thanks for the elaboration, @Jan200101.